### PR TITLE
PHPLIB-650 Run versioned API tests with acceptApiVersion2

### PR DESCRIPTION
**Pull request by alcaeus**
PHPLIB-650

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -356,7 +356,7 @@ tasks:
         - func: "run atlas data lake test"
 
     - name: "test-requireApiVersion"
-      tags: ["versioned_api"]
```
Comment (jmikola) [inline .evergreen/config.yml:359]: Ah, I didn't realize Evergreen supported tags. Was this just never used?

File: .evergreen/config.yml [hunk 2]
```diff
@@ -356,7 +356,7 @@ tasks:
...
+      tags: ["versioned-api"]
```
Comment (jmikola) [inline .evergreen/config.yml:359]: Should this test also use `TESTS: "versioned-api"` to select a subset of the PHPUnit test suite, or does it intentionally need to target the entire suite?

File: .evergreen/config.yml (hunk 1 above)
Comment (alcaeus) [reply .evergreen/config.yml:359]: Yep, we never used this. libmongoc makes heavy use of tags in their config.

File: .evergreen/config.yml (hunk 2 above)
Comment (alcaeus) [reply .evergreen/config.yml:359]: The Versioned API spec calls for running the entire test suite against a server with `requireApiVersion` and authentication enabled to ensure that all commands declare an API version, as they can't be checked using command monitoring.

**Pull request merged**